src/dappsExplorer/searchFilter/DAppsExplorerScreenSearchFilter.tsx

```diff
@@ -0,0 +1,387 @@
+import { useTranslation } from 'react-i18next'
+import {
+  RefreshControl,
+  ScrollView,
+  SectionList,
+  SectionListProps,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native'
+import Animated from 'react-native-reanimated'
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { DappExplorerEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import SearchInput from 'src/components/SearchInput'
+import {
+  dappListWithCategoryNamesSelector,
+  dappsCategoriesAlphabeticalSelector,
+  dappsListErrorSelector,
+  dappsListLoadingSelector,
+  dappsListSelector,
+  dappsMinimalDisclaimerEnabledSelector,
+  favoriteDappIdsSelector,
+} from 'src/dapps/selectors'
+import { fetchDappsList } from 'src/dapps/slice'
+import { DappSection, DappV2, DappV2WithCategoryNames } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import DappFilterChip from 'src/dappsExplorer/DappFilterChip'
+import { searchDappList } from 'src/dappsExplorer/searchDappList'
+import FavoriteDappsSection from 'src/dappsExplorer/searchFilter/FavoriteDappsSection'
+import NoResults from 'src/dappsExplorer/searchFilter/NoResults'
+import useDappFavoritedToast from 'src/dappsExplorer/useDappFavoritedToast'
+import useDappInfoBottomSheet from 'src/dappsExplorer/useDappInfoBottomSheet'
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import { currentLanguageSelector } from 'src/i18n/selectors'
+import Help from 'src/icons/Help'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<DappV2, SectionData>>(SectionList)
+
+interface SectionData {
+  data: DappV2WithCategoryNames[]
+  category: string
+}
+
+export function DAppsExplorerScreenSearchFilter() {
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const horizontalScrollView = useRef<ScrollView>(null)
+
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const dappListWithCategoryNames = useSelector(dappListWithCategoryNamesSelector)
+  const language = useSelector(currentLanguageSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
```

Comment:
can we simplify this? it seems like `favoriteDappsById` is only ever used with `dappListWithCategoryNames`, to filter out the favourite dapps from the dapps list. can we just have a selector that does this, so that we don't need both? the benefit would be that we don't make this computation as often.

Reply:
I made the changes in ee9b0761a714c730d5421ef47e1f1caa702a5aab. Replaced 3 selector imports with one!